Makefile

```diff
@@ -75,14 +75,17 @@ VERSION?=1.9.2
 IMAGE_TAG=$(REGISTRY_NAME)/$(IMAGE_NAME)
 IMAGE_TAG_DEV=$(REGISTRY_NAME)/$(IMAGE_NAME):latest-$(shell git rev-parse --short HEAD)
 
-docker: docker-official
+docker: docker-dev
 
-docker-light: export GOOS=linux
-docker-light: export GOARCH=amd64
-docker-light: export CGO_ENABLED=0
-docker-light:
+docker-dev-binary:
+	GOOS=linux \
+	GOARCH=amd64 \
+	CGO_ENABLED=0 \
+	go build -o bin/packer .
 	mkdir -p dist/linux/amd64
 	cp bin/packer dist/linux/amd64
+
+docker-light: docker-dev-binary
```

Comment:
I don’t think docker-light and docker-full need to build a dev binary binary. They should match what we build in the ci. Which is pulling the bin from Hashi releases. Is there a reason you added the dev build to these targets?

Reply:
So, I would agree with you if we look at the CI build, but the docs in the Dockerfile do state that they are built `from the locally generated binary in ./bin/, and from CI-built binaries within CI.`

Given the phrasing, I assumed this would build from a release in CI, and from the dev binary locally, but we can change that if that's irrelevant.

If we use the Makefile for testing Packer locally with Linux/amd64 binaries, I suggest we keep on building `dev`, but maybe change the version/tags for the image, so they don't point to the version in `PRODUCT_VERSION`. What so you think?

Reply:
Are your referring to the following lines in the README?
>The dev docker image should be built for local dev and testing, while the production docker image, release, is built in CI and makes use of CI-built binaries


If so, then the statement "is built in CI and makes use of CI-built binaries" implies that the  binaries that are built by the [HashiCorp build action](https://github.com/hashicorp/packer/actions/runs/5692510398), which are pushed to Artifactory and used build the docker containers. Not the local dev binaries. The dev binary is merely for testing locally and not part of the Packer ci process.

> If we use the Makefile for testing Packer locally with Linux/amd64 binaries, I suggest we keep on building dev, but maybe change the version/tags for the image, so they don't point to the version in PRODUCT_VERSION. What so you think?

This sounds good to me. If by we keep building dev you mean that the Makefile has a target for it. I would not build them as part of the Packer release/pull-request workflows that run when we merge to main.

Reply:
Seeing as the release-light and release-full targets are not used by the release workflows should we remove them from the Makefile?

Reply:
Yeah, that's probably a good idea. We can focus on delivering only a `dev` image in the Makefile, and let the CI build the releases.
I'll clean this up